spec/unit/room.spec.js

```diff
@@ -1757,4 +1760,70 @@ describe("Room", function() {
             });
         });
     });
+
+    describe("MSC2228_selfDestructAfter", function() {
+        it("should redact an event with a `self_destruct_after` value in the past immediately", function() {
+            const client = {
+                store: {
+                    replaceEvent: jest.fn(() => Promise.resolve()),
+                    save: jest.fn(() => Promise.resolve()),
+                },
+            };
+            const room = new Room(roomId, client, userA);
+            let callCount = 0;
+            room.on("Room.redaction", function(redactionEvent, room) {
+                callCount += 1;
+                expect(redactionEvent.getType()).toBe("m.room.redaction");
+            });
+            room.addLiveEvents([
+                utils.mkEvent({
+                    type: "m.room.message",
+                    room: roomId, event: true,
+                    content: {
+                        msgtype: "m.text",
+                        body: "TOPSECRET",
+                        [UNSTABLE_MSC2228_SELF_DESTRUCT_AFTER.name]: Date.now()-1000,
+                    },
+                }),
+            ]);
+            const event = room.timeline[0];
+            expect(event.isRedacted()).toBeTruthy();
+            expect(event.getContent().body).toBeFalsy();
+            expect(client.store.replaceEvent).toBeCalled();
+            expect(callCount).toBe(1);
+        });
+        it("should redact an event with a future `self_destruct_after` value after that time", function(done) {
+            const client = {
+                store: {
+                    replaceEvent: jest.fn(() => Promise.resolve()),
+                    save: jest.fn(() => Promise.resolve()),
+                },
+            };
+            const room = new Room(roomId, client, userA);
+            room.addLiveEvents([
+                utils.mkEvent({
+                    type: "m.room.message",
+                    room: roomId, event: true,
+                    content: {
+                        msgtype: "m.text",
+                        body: "TOPSECRET",
+                        [UNSTABLE_MSC2228_SELF_DESTRUCT_AFTER.name]: Date.now()+50,
```

Comment:
I'm wondering if this `50` magic value would be flakey at all if it takes more than 50ms to add the event to the live timeline. It's probably rare though if this is sufficiently big enough.

We can control the `Date` in Jest if we want to be absolute about this.

Would be good to at least comment about the `50` magic value here. Something like, 

> // We chose `50` because it gives a small buffer for us to check that the message is not deleted and then deleted a few moments later.
> // FIXME: This is potentially flakey if it takes more than 50ms for the event to be added to the live timeline

Reply:
good point! I've switched now to `jest.useFakeTimers()` and `jest.runAllTimers()` which works nicely :)